app/models/user.rb

```diff
@@ -122,9 +124,14 @@ def all_hooks
     all
   end
 
+  def hide_mfa
+    @hide_mfa.nil? ? true : @hide_mfa
```

Comment:
I'm a bit cautious with merging this change. I can see the inspiration from the `hide_email` attribute, a toggle that the user can control on their profile page that's also persisted in the database. But `hide_mfa` feels very different. It's an internal switch to decide what fields to render in the controller.

Could we instead add an optional boolean argument to `payload` that renders the MFA and any future "private" field? Something like `payload(sensitive_fields: false)` ?

Thoughts?

Reply:
Sure, though if we want to keep using `as_json`, `to_xml`, etc to call payload instead of accessing that directly in the controller, I've had to make an update to our yaml renderer because it doesn't typically pass along arguments. I could call `payload` directly but it seemed from our previous discussion like you wanted to continue rendering the User object and have it behave appropriately as opposed to calling the payload hash directly